nautobot/docs/user-guide/administration/configuration/required-settings.md

```diff
@@ -122,123 +16,93 @@ to different Redis instances/databases per feature.
-
-This is a secret, random string used to assist in the creation new cryptographic hashes for passwords and HTTP cookies. The key defined here should not be shared outside of the configuration file. `SECRET_KEY` can be changed at any time, however be aware that doing so will invalidate all existing sessions.
+<!-- markdownlint-disable blanks-around-lists -->
 
-`SECRET_KEY` should be at least 50 characters in length and contain a random mix of letters, digits, and symbols.
+{% for property, attrs in settings_schema.properties.items() if attrs.is_required_setting|default(false) %}
 
-!!! note
-    A unique `SECRET_KEY` is generated for you automatically when you use `nautobot-server init` to create a new `nautobot_config.py`.
-
-You may run `nautobot-server generate_secret_key` to generate a new key at any time.
-
-```no-highlight
-nautobot-server generate_secret_key
-```
-
-Sample output:
+---
 
-```no-highlight
-+$_kw69oq&fbkfk6&q-+ksbgzw1&061ghw%420u3(wen54w(m
-```
+## `{{ property }}`
 
-Alternatively use the following command to generate a secret even before `nautobot-server` is runnable:
+{% if attrs.version_added|default(None) %}
++++ {{ attrs.version_added }}
+{% endif %}
 
-```no-highlight
-LC_ALL=C tr -cd '[:lower:][:digit:]!@#$%^&*(\-_=+)' < /dev/urandom | fold -w50 | head -n1
-```
+{% if attrs.default_literal|default(None) %}
+**Default:**
 
-Example output:
+{{ attrs.default_literal }}
+{% else %}
+{% with default = attrs.default|default(None) %}
+**Default:**
+{% if default is string %}`"{{ default }}"`
+{% elif default is boolean %}`{{ default|title }}`
+{% elif default is mapping and default != {} %}
 
-```no-highlight
-9.V$@Kxkc@@Kd@z<a/=.J-Y;rYc79<y@](9o9(L(*sS)Q+ud5P
+```python
+{{ default|pprint }}
 ```
 
-!!! warning
-    In the case of a highly available installation with multiple web servers, `SECRET_KEY` must be identical among all servers in order to maintain a persistent user session state.
-
-For more details see [Nautobot Configuration](index.md).
+{% else %}`{{ default }}`
+{% endif %}
+{% endwith %}
+{% endif %}
+
+{% if attrs.enum|default(None) %}
+**Permitted Values:**
+
+{% for enum in attrs.enum %}
+* `{{ enum|pprint }}`
+{% endfor %}
+{% endif %}
+
+{% if attrs.environment_variable|default(None) %}
+**Environment Variable:** `{{ attrs.environment_variable }}`
+{% elif attrs.properties|default(None) != None and attrs.properties.default|default(None) != None %}
+{% for property_attrs in attrs.properties.default.properties.values() if property_attrs.environment_variable|default(None) %}
+{% if loop.first %}
+**Environment Variables:**
+
+{% endif %}
+* `{{ property_attrs.environment_variable }}`
+{% endfor %}
+{% elif attrs.properties|default(None) != None %}
+{% for property_attrs in attrs.properties.values() if property_attrs.environment_variable|default(None) %}
+{% if loop.first %}
+**Environment Variables:**
+
+{% endif %}
+* `{{ property_attrs.environment_variable }}`
+{% endfor %}
+{% endif %}
```

Comment:
Wonder if we can/should extract the common Jinja template from `optional-settings` and `required-settings` into a reusable fragment with `mkdocs-include-markdown-plugin`?

Reply:
https://github.com/nautobot/nautobot/pull/5363/commits/31ab02d8ec81f341472cd322690c20a34caf4841

If this looks ok to you, I'll approve and we can merge.

Reply:
:shipit: